tests/end_to_end/test_target_bigquery.py

```diff
@@ -0,0 +1,308 @@
+                                 ' WHERE address_id = 1')
+        #  FULL_TABLE
+        self.run_query_tap_mysql('DELETE FROM no_pk_table WHERE id > 10')
+
+        # 3. Run tap second time - both fastsync and a singer should be triggered, there are some FULL_TABLE
+        assertions.assert_run_tap_success(tap_mariadb_id, TARGET_ID, ['fastsync', 'singer'])
+        assertions.assert_row_counts_equal(self.run_query_tap_mysql, self.run_query_target_bigquery)
+        assertions.assert_all_columns_exist(self.run_query_tap_mysql, self.e2e.run_query_target_bigquery,
+                                            mysql_to_bigquery.tap_type_to_target_type, {'blob_col'})
+
+        # Checking if mask-date transformation is working
+        result = self.run_query_target_bigquery(
+            'SELECT count(1) FROM ppw_e2e_tap_mysql.address '
+            'where EXTRACT(MONTH FROM date_created) != 1 or EXTRACT(DAY FROM date_created) != 1;')[0][0]
+
+        assert result == 0
+
+        # Checking if conditional MASK-NUMBER transformation is working
+        result = self.run_query_target_bigquery(
+            'SELECT count(1) FROM ppw_e2e_tap_mysql.address '
+            'where zip_code_zip_code_id != 0 and REGEXP_CONTAINS(street_number, \'[801]\');')[0][0]
+
+        assert result == 0
+
+        # Checking if conditional SET-NULL transformation is working
+        result = self.run_query_target_bigquery(
+            'SELECT count(1) FROM ppw_e2e_tap_mysql.edgydata '
+            'where "GROUP" is not null and "CASE" = \'B\';')[0][0]
+
+        assert result == 0
+
+    @pytest.mark.dependency(depends=['import_config'])
+    def test_resync_mariadb_to_bq(self, tap_mariadb_id=TAP_MARIADB_ID):
+        """Resync tables from MariaDB to Bigquery"""
+        assertions.assert_resync_tables_success(tap_mariadb_id, TARGET_ID, profiling=True)
+        assertions.assert_row_counts_equal(self.run_query_tap_mysql, self.run_query_target_bigquery)
+        assertions.assert_all_columns_exist(self.run_query_tap_mysql, self.run_query_target_bigquery,
+                                            mysql_to_bigquery.tap_type_to_target_type)
+
+    # pylint: disable=invalid-name
+    @pytest.mark.dependency(depends=['import_config'])
+    def test_resync_mariadb_to_bq_with_split_large_files(self, tap_mariadb_id=TAP_MARIADB_SPLIT_LARGE_FILES_ID):
+        """Resync tables from MariaDB to Bigquery using splitting large files option"""
+        assertions.assert_resync_tables_success(tap_mariadb_id, TARGET_ID, profiling=True)
+        assertions.assert_row_counts_equal(self.run_query_tap_mysql, self.run_query_target_bigquery)
+        assertions.assert_all_columns_exist(self.run_query_tap_mysql, self.run_query_target_bigquery,
+                                            mysql_to_bigquery.tap_type_to_target_type)
+
+    # pylint: disable=invalid-name
+    def test_resync_pg_to_bq_with_split_large_files(self, tap_postgres_id=TAP_POSTGRES_SPLIT_LARGE_FILES_ID):
```

Comment:
This test is always running but should run only when it's configured. Can you please annotate this test with:
```
@pytest.mark.dependency(depends=['import_config'])
```